### DISCUSSION TITLE
Ratings from Metadata (Vorbiscomment) Proposal

### DISCUSSION THREAD

**Discussion by sammypanda** [navidrome/navidrome#2928] [Ideas]
# Introduction
Currently, Navidrome does not inherit song ratings from the metadata. In this discussion, I'll propose implementing support for rating metadata.

# Background
Song ratings already exist in Navidrome and is a crucial aspect of music organisation for many users. They provide a quick and effective way to categorize and prioritize songs based on personal preferences. Currently, many music files contain metadata tags, which include rating information. However, Navidrome does not align it's ratings with this existing data. This results in using hacky solutions to try to transfer ratings over.

# Proposal
I propose implementing the rating part of the Vorbis comment standard for Navidrome. I also propose for Navidrome to embed Vorbis comment ratings when set on the interface, but this may be considered a lesser priority.

The most commonly used *informal* standard for Vorbis comment ratings is:
- Field name "rating"
- Integer 0-100.

for example:
```
rating=80
```

# Requirements
## Metadata Parsing: 
Functionality to parse Vorbis comment metadata embedded within music files to extract rating information.
## Metadata embedding <?> 
Writing to the file to embed the rating metadata that has been set on the user interface. If this could be implemented, it likely should be done with a library.

# Conclusion
Implementing the 'Vorbis comment' standard rating metadata would greatly enhance the functionality and usability of Navidrome. I feel it aligns with the project's goal of reading and using all the curated metadata. It would be a valuable addition to the application and benefit any users who rely on song ratings for organizing their music collections.

# References/Read More
'Vorbis comment' summary: https://handwiki.org/wiki/Vorbis_comment
Tag specifications: https://wiki.hydrogenaud.io/index.php?title=Tag_Mapping

**Comment by metalheim** [comment]
Its not implemented for a few reasons:
1. Navidrome doesn't write to the files (metadata or anything), It's strictly read-only
2. That will create conflicts that are Impossible to resolve (rating is read from file, but then changed in navidrome, then the files rating is changed - what should navidrome respect?)
3. Navidrome is multi-user. The rating in a files metadata is 1 users opinon that is certainly not valid for every user. 

The only way i see "rating from file metadata" implemented is a "light" version: 
1. Only read from file
2. store it in an additional field (not the normal "rating" field)
3. don't allow changes via frontend. 

That would circumvent all the issues mentioned above pretty much. It's a bit less useless as ratings arent synced back to the files, but still somewhat useful if you already have most of your files :)

**Reply by sammypanda** [reply]
That light version would make sense to me. Given it could be used to sort/for smart playlists.

**Reply by loosenut** [reply]
I would love to see this.

If adding a read mode is out the question, changing the rating on the front-end could still be allowed. Just give priority to the rating assigned in Navidrome.
